docs/library/ussl.rst

```diff
@@ -46,3 +66,40 @@ Constants
           ussl.CERT_REQUIRED
 
     Supported values for *cert_reqs* parameter.
+
+    - CERT_NONE: in client mode accept just about any cert, in server mode do not
+      request a cert from the client.
+    - CERT_OPTIONAL: in client mode behaves the same as CERT_REQUIRED and in server
+      mode requests an optional cert from the client for authentication.
+    - CERT_REQUIRED: in client mode validates the server's cert and
+      in server mode requires the client to send a cert for authentication. Note that
+      ussl does not actually support client authentication.
+
+Pre-Shared Key (PSK) cipher suites
+----------------------------------
+
+TLS supports authentication and encryption using a pre-shared key (i.e. a
+key that both client and server know) as an alternative to the public
+key cryptography commonly used on the web for HTTPS.  PSK is starting
+to be used for MQTT, e.g. in mosquitto, to simplify the set-up and avoid
+having to go through the whole CA, cert, and private key process.
+
+A pre-shared key is a binary string of up to 32 bytes and is commonly
+represented in hex form. In addition to the key, clients can also
+present an id/ident and typically the server allows a different key
+to be associated with each client id. In effect this is very similar
+to username and password pairs, except that unlike a password the
+key is not directly transmitted to the server, thus a connection to a
```

Comment:
I hope that in today's times most services don't transmit passwords directly :) But instead only a salted hash or something similar.

The strength of TLS-PSK (unlike standard password authentication) is that the PSK (or any part/form of it) will never be compromised, so a malicious server talking to a valid client can't deduce any useful information about  it. This should be noted here (it's one of the strengths of this mode, you don't need to care about malicious servers/clients either way)

Reply:
Good point, will add!